include/roboteam_ai/treeinterp/BTFactory.h

```diff
@@ -38,7 +38,9 @@ class BTFactory {
 
     static bt::BehaviorTree::Ptr getKeeperTree();
 
-    static void setCurrentTree(const std::string &currentTree);
+    static void setCurrentTree(std::shared_ptr<rtt::ai::analysis::Play> play);
```

Comment:
Is shared_ptr necessary? Or can you just use reference to unique_ptr?

Reply:
As in what is your ownershipmodel: does the BTFactory own the trees or are they owned somewhere else as well?

Reply:
The plays vector is owned by a PlayChecker instance that stays alive for the program duration. I currently made it so that this is a vector of unique pointers. Whenever I need them, I pass the internal pointer from the unique pointers in the vector. Therefore the resource is only ever owned by 1 thing. Not sure if this is convoluted